pytorch_lightning/loggers/wandb.py

```diff
@@ -159,12 +152,14 @@ def experiment(self) -> Run:
                 **self._kwargs
             ) if wandb.run is None else wandb.run
 
-            # offset logging step when resuming a run
-            self._step_offset = self._experiment.step
-
             # save checkpoints in wandb dir to upload on W&B servers
             if self._save_dir is None:
                 self._save_dir = self._experiment.dir
+
+            # define default x-axis (for latest wandb versions)
+            if getattr(self._experiment, "define_metric", None):
+                self._experiment.define_metric("*", step_metric='train/step', step_sync=True)
```

Comment:
would it make sense to have this step metric name as an argument for customization?

Reply:
It could be useful but there would be a few extra things to think of:
* what to do if that metric never appears?
* what is the start value if it's not logged at the beginning (for example if it's an evaluation metric)

If you call this method explicitly you will actually be able to overwrite this default behavior.
Let me see with the W&B team what they plan to do if the step_metric is not monotonic to have an idea of how we should handle these custom cases (and feel free if you have any example in mind).